apis/nucleus/src/components/Cell.jsx

```diff
@@ -320,6 +325,36 @@ const Cell = forwardRef(
     const [bgColor, setBgColor] = useState(undefined);
     const [bgImage, setBgImage] = useState(undefined); // {url: "", size: "", pos: ""}
     const [titleStyles, setTitleStyles] = useState(undefined);
+    const initState = { mouseX: -1, mouseY: -1 };
+    const [contextMenu, setContextMenu] = React.useState(initState);
+    const [menuItems, setMenuItems] = React.useState(Menu());
+
+    const handleContextMenu = (event) => {
+      event.preventDefault();
+      if (state.sn.component) {
+        // I need the object for layout container here to create context menu for the container
+        const menu = contextMenuBuilderForObject(object);
+        const menuBuidler = new MenuBuilder();
+        menuBuidler.registerBuilder('object', contextMenuBuilderForObject);
+        state.sn.component.onContextMenu(menu, event, menuBuidler);
+        setMenuItems(menu);
+        setContextMenu(
+          contextMenu.mouseX === -1
+            ? {
+                mouseX: event.clientX + 2,
+                mouseY: event.clientY - 6,
+              }
+            : // repeated contextmenu when it is already open closes it with Chrome 84 on Ubuntu
+              // Other native context menus might behave different.
+              // With this behavior we prevent contextmenu from the backdrop to re-locale existing context menus.
+              initState
+        );
+      }
+    };
+
+    const closeContextMenu = () => {
+      setContextMenu(initState);
+    };
```

Comment:
in general, try to move code out of Cell.jsx so it doesn't grow too much.
Also the "object" here should not be needed, should be something else.

Reply:
Yes. I have moved them out. This is just a quick test. Please don't review this for now since it not ready yet. This is one messy :).